### PR TITLE
DDP-8746 . update osteo patch 

### PR DESCRIPTION
**Pull request by ssettipalli**
update osteo patch to cleanup incomplete ABOUTYOU/ABOUTCHILD
Just added new activity codes to existing activity codes

Added new events to create ABOUTYOU/ABOUTCHILD on re-consent submission to handle ABOUTYOU edge case

NOTE: 
Patch is targeted for prod. Executing the patch against current Dev/Test/Staging will cleanup even new OSTEO2 users activity instances if in created or in-progress state. 
"RELEASE_SELF", "RELEASE_MINOR", "ABOUTYOU", "ABOUTCHILD"

limitation:
The approach works only in scenarios where release is not yet submitted . 
we might need to manually create ABOUTYOU instance for users who already submitted release.
Manually run sql to find these ptps and manually create ABOUTYOU instances
OR add a event to create ABOUTYOU if not exists on re-consent (went with adding new events approach)

If  we want to run in Dev/Test/Staging, might need to alter queries to look for users / activity-instances created from some old date.

QA: Tested in Dev and Test by commenting out actual delete and verifying that right activity-instances are pulled.
Crosse checked using this SQL too
select
u.guid,
u.hruid,
inst.activity_instance_guid,inst.activity_instance_id,
st.activity_instance_status_type_code
from
study_activity a,
activity_instance inst,
activity_instance_status stat,
activity_instance_status_type st,
user u
where
u.user_id = inst.participant_id
and
st.activity_instance_status_type_id = stat.activity_instance_status_type_id
and
stat.activity_instance_id = inst.activity_instance_id
and
inst.study_activity_id = a.study_activity_id
and
a.study_activity_id in (17387,17388)
and
stat.updated_at = (select max(stat2.updated_at) from activity_instance_status stat2 where stat2.activity_instance_id = stat.activity_instance_id)
and
st.activity_instance_status_type_code != 'COMPLETE';

### CODE REVIEW

**Review by MocanaAtBroad — DISMISSED** [review]

**Review by MocanaAtBroad — DISMISSED** [review]

**Review by MocanaAtBroad — DISMISSED** [review]

File: pepper-apis/studybuilder-cli/studies/osteo/study-events.conf [hunk 1]
```diff
@@ -493,6 +493,178 @@
...
+    ##CONSENT_ASSENT to ABOUTYOU if RELEASE_MINOR is submitted.. handle osteo2 edge case
+    {
+      "trigger": {
+        "type": "ACTIVITY_STATUS",
+        "activityCode": "CONSENT_ASSENT",
+        "statusType": "COMPLETE"
+      },
+      "action": {
+        "type": "ACTIVITY_INSTANCE_CREATION",
+        "activityCode": "ABOUTYOU"
```
Comment (ssettipalli) [inline pepper-apis/studybuilder-cli/studies/osteo/study-events.conf:561]: @LKapanad : I thin this should be ABOUTCHILD and NOT ABOUTYOU !
From CONSENT_ASSENT -> ABOUTCHILD, right ?

File: pepper-apis/studybuilder-cli/studies/osteo/study-events.conf [hunk 2]
```diff
@@ -493,6 +493,178 @@
...
+    ##PARENTAL_CONSENT to ABOUTYOU if RELEASE_MINOR is submitted.. handle osteo2 edge case
+    {
+      "trigger": {
+        "type": "ACTIVITY_STATUS",
+        "activityCode": "PARENTAL_CONSENT",
+        "statusType": "COMPLETE"
+      },
+      "action": {
+        "type": "ACTIVITY_INSTANCE_CREATION",
+        "activityCode": "ABOUTYOU"
```
Comment (ssettipalli) [inline pepper-apis/studybuilder-cli/studies/osteo/study-events.conf:526]: If we did PARENTAL_CONSENT, isn't it pediatric and we should create "ABOUTCHILD" ?

File: pepper-apis/studybuilder-cli/studies/osteo/study-events.conf [hunk 3]
```diff
@@ -493,6 +493,178 @@
...
+    ##PEDIATRIC ADDENDUM to ABOUTYOU V1->V2
+    {
+      "trigger": {
+        "type": "ACTIVITY_STATUS",
+        "activityCode": "CONSENT_ADDENDUM_PEDIATRIC",
+        "statusType": "COMPLETE"
+      },
+      "action": {
+        "type": "ACTIVITY_INSTANCE_CREATION",
+        "activityCode": "ABOUTYOU"
```
Comment (ssettipalli) [inline pepper-apis/studybuilder-cli/studies/osteo/study-events.conf:597]: I thin this should be ABOUTCHILD since its from CONSENT_ADDENDUM_PEDIATRIC

File: pepper-apis/studybuilder-cli/studies/osteo/study-workflows.conf [hunk 4]
```diff
@@ -171,6 +171,51 @@
...
+        {
+          "type": "ACTIVITY",
+          "activityCode": "RELEASE_SELF",
+          "expression": """user.studies["CMI-OSTEO"].forms["RELEASE_SELF"].hasInstance()"""
+        }
+      ]
+    },
+    {
+      "from": {
+        "type": "ACTIVITY"
```
Comment (ssettipalli) [inline pepper-apis/studybuilder-cli/studies/osteo/study-workflows.conf:194]: I believe this flow TO can be added to above one at the end and get rid of this one.
Same for below one too but doesn't hurt to have diff one

File: pepper-apis/studybuilder-cli/studies/osteo/study-workflows.conf [hunk 5]
```diff
@@ -291,6 +414,33 @@
...
     },
+    {
+      "from": {
+        "type": "ACTIVITY",
+        "activityCode": "CONSENT_ASSENT"
+      },
+      "to": [
+        {
+          "type": "ACTIVITY",
+          "activityCode": "ABOUTYOU",
```
Comment (ssettipalli) [inline pepper-apis/studybuilder-cli/studies/osteo/study-workflows.conf:425]: Isn't it ABOUTCHILD from CONSENT_ASSENT ?

File: pepper-apis/studybuilder-cli/studies/osteo/study-events.conf (hunk 1 above)
Comment (ssettipalli) [reply pepper-apis/studybuilder-cli/studies/osteo/study-events.conf:561]: Didn't realize that ABOUTCHILD doesn't exist in OS2

**Review by LKapanad — APPROVED** [review]

**Review by MocanaAtBroad — APPROVED** [review]
